stage0/run.go

```diff
@@ -382,17 +382,17 @@ func prepareStage1Image(cfg PrepareConfig, img types.Hash, cdir string, useOverl
 		return fmt.Errorf("error creating stage1 directory: %v", err)
 	}
 
-	if useOverlay {
-		if err := cfg.Store.RenderTreeStore(img.String(), false); err != nil {
+	if err := cfg.Store.RenderTreeStore(img.String(), false); err != nil {
+		return fmt.Errorf("error rendering tree image: %v", err)
+	}
+	if err := cfg.Store.CheckTreeStore(img.String()); err != nil {
+		log.Printf("Warning: tree cache is in a bad state. Rebuilding...")
+		if err := cfg.Store.RenderTreeStore(img.String(), true); err != nil {
 			return fmt.Errorf("error rendering tree image: %v", err)
 		}
-		if err := cfg.Store.CheckTreeStore(img.String()); err != nil {
-			log.Printf("Warning: tree cache is in a bad state. Rebuilding...")
-			if err := cfg.Store.RenderTreeStore(img.String(), true); err != nil {
-				return fmt.Errorf("error rendering tree image: %v", err)
-			}
-		}
-	} else {
+	}
+
+	if !useOverlay {
```

Comment:
to make sure I understand this correctly: when not using overlayfs, we are rendering the stage1 once in the tree cache, just so we later have access to `enter`, but then we are expanding it again from scratch into the actual container directory again here?